libs/payments/paypal/src/lib/paypal.service.ts

```diff
@@ -4,8 +4,39 @@
 
 import { Injectable } from '@nestjs/common';
 import { PayPalManager } from './paypal.manager';
+import { PaypalCustomerManager } from './paypalCustomer/paypalCustomer.manager';
+import { CurrencyManager } from '@fxa/payments/currency';
 
 @Injectable()
 export class PayPalService {
-  constructor(private paypalManager: PayPalManager) {}
+  constructor(
+    private paypalManager: PayPalManager,
+    private paypalCustomerManager: PaypalCustomerManager,
+    private currencyManager: CurrencyManager
+  ) {}
+
+  /**
+   * Create and verify a billing agreement is funded from the appropriate
+   * country given the currency of the billing agreement.
+   */
+  async createBillingAgreement(uid: string, token: string, currency: string) {
```

Comment:
This should likely be a manager method, rather than a service-level method given the structure docs that Reino has recently posted.

Reply:
PaypalManager would be a good location for this.

Reply:
As per our Slack conversation, overwriting method in PayPalManager and updating the test suite to include providers and factories.